salt/proxy/restconf.py

```diff
@@ -0,0 +1,226 @@
+"""
+Proxy Minion to manage Restconf Devices
+
+:codeauthor: Jamie (Bear) Murphy <jamiemurphyit@gmail.com>
+:maturity:   new
+:platform:   any
+
+Usage
+=====
+
+.. note::
+
+    To be able to use this module you need to enable to RESTCONF on your device
+    and have https enabled.
+
+    Cisco Configuration example:
+
+    .. code-block:: bash
+
+        switch# conf t
+        switch(config)# restconf
+        switch(config)# ip http secure-server
+
+.. note::
+
+    RESTCONF requires modern OS distributions.
+    This plugin has been written specifically to use JSON Restconf endpoints
+
+Pillar
+------
+
+The ``restconf`` proxy configuration requires the following parameters in order
+to connect to the network switch:
+
+transport: ``https`` (str)
+    Specifies the type of connection transport to use. Valid values for the
+    connection are ``https``, and  ``http``.
+    http is not part of the restconf standard but is included as an option here
```

Comment:
Yes. I talked about the in a previous comment too and added it to that description block. 
Unfortunately not all network device manufacturers are following the rule correctly and some ONLY have the ability to have an insecure http server. I have already experienced this during dev of this code

Reply:
Sorry, I guess it wasn't clear - I was suggesting that be reworded as the quote. It would probably be worth linking to the standard generally, somewhere in the docs (maybe module and proxy? Certainly at least in the proxy)

Reply:
my bad. fixed in 44aa4e6abc581a9d4f4b717cc70810e72e8dfb12